### PR TITLE
feat: make data gathering more robust

### PR DESCRIPTION
**Pull request by eggfoobar**
added retry logic to lesson chance of flakes during runs 
removed custom exec logic and leveraged framework pod package to execute commands on pods

### CODE REVIEW

File: test/extended/cpu_partitioning/crio.go [hunk 1]
```diff
@@ -276,63 +275,68 @@ func collectContainerInfo(ctx context.Context, cfg *rest.Config, c kubernetes.In
...
+		PreserveWhitespace: false,
 	}
 
 	info := []crioContainerData{}
-	if err := json.Unmarshal(out, &info); err != nil {
-		return nil, fmt.Errorf("error parsing container output to json: %w", err)
-	}
+	err = retry.OnError(backoff, shouldRetryExec,
+		func() error {
+			out, _, execErr := e2epod.ExecWithOptions(oc.KubeFramework(), execOptions)
```
Comment (ffromani) [inline test/extended/cpu_partitioning/crio.go:309]: the retry logic seems nice, but why using this code is better than the previous code? just avoiding code duplication or other reasons?

Comment (eggfoobar) [reply test/extended/cpu_partitioning/crio.go:309]: yeah, noticed this had some nicer options (like logging the request) and figured we could avoid code duplications and hopefully less things for us to manage and worry about

File: test/extended/cpu_partitioning/crio.go [hunk 2]
```diff
@@ -276,63 +275,68 @@ func collectContainerInfo(ctx context.Context, cfg *rest.Config, c kubernetes.In
 	}
 
 	pod := &pods.Items[0]
+	podName := pod.Name
+	podNamespace := pod.Namespace
+	containerName := pod.Spec.Containers[0].Name
```
Comment (ffromani) [inline test/extended/cpu_partitioning/crio.go:280]: not sure we need these local vars, but it won't hurt either

**Review by ffromani — COMMENTED** [review]
/lgtm

**Pull request merged**